azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -144,6 +177,25 @@ func resourceArmKubernetesClusterNodePool() *schema.Resource {
 				}, false),
 			},
 
+			"priority": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  string(containerservice.Regular),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(containerservice.Regular),
+					string(containerservice.Spot),
+				}, false),
+			},
+
+			"spot_max_price": {
```

Comment:
This name is inconsistent with `windows_virtual_machine_scale_set` and `linux_virtual_machine_scale_set`

Reply:
unfortunately different teams within Microsoft can refer to the same functionality in different ways - as such unfortunately we need to use the name being exposed by AKS here (which is `spotMaxPrice` - rather than "spot bid price" used by the VM Scale Set API's)